tests/test_leftovers.py

```diff
@@ -0,0 +1,294 @@
+    # post two records of different size
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["size"] = 100
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?size=" + str(data["size"]))
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper size
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["size"] == 100
+
+
+def test_index_list_by_file_name(client, user):
+    # post two records of different name
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["file_name"] = "test_file"
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?file_name=" + data["file_name"])
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper name
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["file_name"] == data["file_name"]
+
+
+def test_index_list_by_authz(client, user):
+    # post two records of different authz
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["authz"] = ["test_authz"]
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?authz=" + data["authz"][0])
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper authz
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["authz"] == data["authz"]
+
+
+def test_index_list_by_version(client, user):
+    # post two records of different version
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    data["version"] = "3"
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    res = client.get("/index/?version=" + data["version"])
+    assert res.status_code == 200
+    rec = res.json
+    # assert only one record returned and returned with proper version
+    assert len(rec["records"]) == 1
+    assert rec["records"][0]["version"] == data["version"]
+
+
+def test_negate_filter_file_name(client, user):
+    # post two records of different file name
+    data1 = get_doc()
+    data1["file_name"] = "test_file_name_1"
+    res_1 = client.post("/index/", json=data1, headers=user)
+    assert res_1.status_code == 200
+
+    data2 = get_doc()
+    data2["file_name"] = "test_file_name_2"
+    res_2 = client.post("/index/", json=data2, headers=user)
+    assert res_2.status_code == 200
+
+    negate_param = {"file_name": "test_file_name_2"}
+    res = client.get("/index/?negate_params=" + json.dumps(negate_param))
```

Comment:
sort of the same comment here; could you add a 3rd record that has _no_ file name? from the docs, i think this endpoint should also return records that don't have the "negate param"

Reply:
I believe it returns either the negate param or the blank keys for that param